consensus/pbft/batch.go

```diff
@@ -128,18 +127,13 @@ func (op *obcBatch) Close() {
 
 func (op *obcBatch) submitToLeader(req *Request) events.Event {
 	// Broadcast the request to the network, in case we're in the wrong view
-	op.broadcastMsg(&BatchMessage{&BatchMessage_Request{req}})
-
+	op.broadcastMsg(&BatchMessage{Payload: &BatchMessage_Request{Request: req}})
 	op.logAddTxFromRequest(req)
 	op.reqStore.storeOutstanding(req)
 	op.startTimerIfOutstandingRequests()
-
-	// if we believe we are the leader, then process this request
-	leader := op.pbft.primary(op.pbft.view)
-	if leader == op.pbft.id && op.pbft.activeView {
+	if op.pbft.primary(op.pbft.view) == op.pbft.id && op.pbft.activeView {
```

Comment:
For me (with no so much knowledge about this part of the code), the former version was easier to understand.